indico/modules/events/surveys/controllers/display.py

```diff
@@ -88,8 +93,9 @@ def _process(self):
         form = self._make_form()
         if form.validate_on_submit():
             submission = self._save_answers(form)
-            if submission.is_anonymous:
+            if submission.is_anonymous and submission.user:
                 submission.user = None
+                AnonymousSurveySubmission(survey=self.survey, user=session.user)
```

Comment:
```suggestion
                self.survey.anonymous_submissions.append( AnonymousSurveySubmission(user=session.user))
```

This is mostly a matter of preference, I just find it a bit weird to instantiate a class w/o doing something with it, so I prefer to use one of the relationships like this..